test-install workflow broken

See e.g. https://github.com/aiidateam/aiida-core/actions/runs/3762763338/jobs/6395771660

Only two errors of the test suite fail
```
FAILED tests/orm/test_querybuilder.py::TestRepresentations::test_as_sql_literal_quote - sqlalchemy.exc.CompileError: No literal value renderer is available for literal value "['Si']" with datatype JSONB
FAILED tests/orm/test_querybuilder.py::test_analyze_query - sqlalchemy.exc.CompileError: No literal value renderer is available for literal value "['x', 1]" with datatype JSONB
```

Given that the only difference to the testing on `main` should be the dependencies that are installed, I suspect one of the packages is responsible - e.g. on python 3.8 this installs `SQLAlchemy-1.4.45-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl` while we're pinning at SQLAlchemy==1.4.27

I don't know what is causing this, though - if someone could have a look that would be great 

@sphuber @aiidateam/dependency-manager

--- a/aiida/storage/psql_dos/orm/querybuilder/main.py
+++ b/aiida/storage/psql_dos/orm/querybuilder/main.py
@@ -20,7 +20,7 @@ from sqlalchemy import func as sa_func
 from sqlalchemy import not_, or_
 from sqlalchemy.dialects.postgresql import JSONB
 from sqlalchemy.engine import Row
-from sqlalchemy.exc import SAWarning
+from sqlalchemy.exc import CompileError, SAWarning
 from sqlalchemy.orm import aliased
 from sqlalchemy.orm.attributes import InstrumentedAttribute, QueryableAttribute
 from sqlalchemy.orm.query import Query
@@ -889,7 +889,8 @@ def compile_query(query: Query, literal_binds: bool = False) -> SQLCompiler:
 
             try:
                 return super().render_literal_value(value, type_)
-            except NotImplementedError:
+            # sqlalchemy<1.4.45 raises NotImplementedError, sqlalchemy>=1.4.45 raises CompileError
+            except (NotImplementedError, CompileError):
                 if isinstance(value, list):
                     values = ','.join(self.render_literal_value(item, type_) for item in value)
                     return f"'[{values}]'"

